Makefile

```diff
@@ -237,19 +220,22 @@ else
 	CGO_ENABLED=0 go build -v -i -ldflags '${LDFLAGS} -extldflags -static' -o $@ ./cmd/argoexec
 endif
 
-.PHONY: executor-image
-executor-image: dist/argoexec.image
+argoexec-image:
+argoexec-dev-image: dist/argoexec
+	docker tag $(IMAGE_NAMESPACE)/argoexec-dev:$(VERSION) $(IMAGE_NAMESPACE)/argoexec:$(VERSION)
 
-ifeq ($(DEV_IMAGE),true)
-dist/argoexec.image: dist/argoexec
-	[ -e argoexec ] || mv dist/argoexec .
-	$(call docker_build,argoexec-dev,argoexec)
-	mv argoexec dist/
-else
-dist/argoexec.image: $(ARGOEXEC_PKGS) go.sum
-	$(call docker_build,argoexec,argoexec)
-endif
-	touch dist/argoexec.image
+%-image:
+	[ ! -e dist/$* ] || mv dist/$* .
+	docker build -t $(IMAGE_NAMESPACE)/$*:$(VERSION) --target $* .
+	[ ! -e $* ] || mv $* dist/
+	docker run --rm -t $(IMAGE_NAMESPACE)/$*:$(VERSION) version
+	if [ $(K3D) = true ]; then k3d image import $(IMAGE_NAMESPACE)/$*:$(VERSION); fi
+	if [ $(DOCKER_PUSH) = true ] && [ $(IMAGE_NAMESPACE) != argoproj ] ; then docker push $(IMAGE_NAMESPACE)/$*:$(VERSION) ; fi
+
+scan-images: scan-workflow-controller scan-argoexec scan-argocli
+
+scan-%:
+	docker scan --severity=high $(IMAGE_NAMESPACE)/$*:$(VERSION)
```

Comment:
@simster7 @sarabala1979 latest version of `docker` has `docker scan` which runs Synk.